app/models/agreements/iaa.rb

```diff
@@ -0,0 +1,43 @@
+# Abstract class to combine IaaGtcs and IaaOrders for JSON serialization
+module Agreements
+  class Iaa
+    include ActiveModel::Model
+
+    attr_accessor :gtc, :order
+
+    delegate :gtc_number, to: :gtc
+    delegate :order_number, to: :order
+
+    def iaa_number
+      "#{gtc.gtc_number}-#{'%04d' % order.order_number}-#{'%04d' % order.mod_number}"
+    end
+
+    def partner_account
+      gtc.partner_account.requesting_agency
+    end
+
+    def gtc_status
+      gtc.partner_status
+    end
+
+    def order_status
+      order.partner_status
+    end
+
+    def ==(other)
+      other.gtc == gtc && other.order == order
+    end
+
+    def method_missing(method)
```

Comment:
I think this may be a good place for a comment.

Also, `#send` is a little scary to me. Could we reduce the number of method calls we can send to the underlying objects? Maybe an allowlist of allowed methods?

Reply:
completely refactored using `delegate` and updated the specs using shoulda matchers. 🤦